services/QuillLMS/client/app/bundles/Teacher/components/classrooms/active_classrooms.tsx

```diff
@@ -101,19 +102,61 @@ export default class ActiveClassrooms extends React.Component<ActiveClassroomsPr
     if (showModal || selectedClassroomId) {
       this.setState({ showModal, selectedClassroomId })
     }
+  }
+
+  getCleverClassrooms = () => {
+    const { user } = this.props
+    const { clever_id } = user
+    const { attemptedImportCleverClassrooms } = this.state
+
+    if (clever_id) {
+      this.setState({ cleverClassroomsLoading: true}, () => {
+        requestGet('/clever_integration/teachers/retrieve_classrooms', body => {
+          this.setState({ attemptedImportCleverClassrooms: true })
+
+          if (body.reauthorization_required) {
+            this.openModal(reauthorizeCleverModal)
+          }
+          else if (body.quill_retrieval_processing) {
+            this.initializePusherForCleverClassrooms(body.user_id)
+          } else {
+            const { classrooms_data, existing_clever_ids } = body
+            const { classrooms } = classrooms_data
+            const cleverClassrooms = classrooms.filter(classroom => !existing_clever_ids.includes(classroom.clever_id))
+            const newStateObj: any = { cleverClassrooms, cleverClassroomsLoading: false }
+
+            if (attemptedImportCleverClassrooms) {
+              newStateObj.attemptedImportCleverClassrooms = false
+              this.setState(newStateObj, this.clickImportCleverClassrooms)
+            } else {
+              this.setState(newStateObj)
+            }
 
+            if (cleverClassrooms.length) {
+              this.openModal(importCleverClassroomsModal)
+            } else {
+              this.openModal(cleverClassroomsEmptyModal)
+            }
+          }
+        })
+      })
+    }
   }
 
-  getGoogleClassrooms() {
-    if (this.props.user.google_id) {
+  getGoogleClassrooms = () => {
+    const { user } = this.props
+    const { clever_id, google_id } = user
+    const { attemptedImportGoogleClassrooms } = this.state
+
+    if (!clever_id && google_id) {
```

Comment:
Just curious about the thinking behind this - theoretically I don't think we should have users that are both Google and Clever users but if we did, we might still want them to access their Google classrooms, right?